File: php/testcode/benchmark_test_01005.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest01005(BenchmarkRequest $req): BenchmarkResponse {
    $userInput = $req->param('data');
    $descriptors = [0 => ['pipe', 'r'], 1 => ['pipe', 'w'], 2 => ['pipe', 'w']];
    $proc = proc_open("sort", $descriptors, $pipes);
    fwrite($pipes[0], $userInput);
    fclose($pipes[0]);
    $output = stream_get_contents($pipes[1]);
    fclose($pipes[1]);
    fclose($pipes[2]);
    proc_close($proc);
    return BenchmarkResponse::ok($output);
}

File: rust/testcode/benchmark_test_01005.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let idx: usize = req.param("index").parse().unwrap_or(0);
    let data = [1u8, 2, 3, 4, 5, 6, 7, 8];
    if idx < data.len() - 1 {
        let val = unsafe { *data.get_unchecked(idx) };
        super::shared::BenchmarkResponse::ok(&format!("val={}", val))
    } else {
        super::shared::BenchmarkResponse::bad_request("OOB")
    }
}
